src/ShallowTraversal.js

```diff
@@ -110,10 +110,13 @@ export function buildPredicate(selector) {
           return node => nodeHasId(node, selector.slice(1));
 
         case SELECTOR.PROP_TYPE: {
-          const propKey = selector.split(/\[([a-zA-Z-]*?)(=|])/)[1];
+          const propKey = selector.split(/\[([a-z]+[a-z-0-9-]*?)(=|])/i);
+          if (propKey.length === 1) {
+            throw new TypeError('Enzyme::Invalid attribute selector');
```

Comment:
It'd be helpful if you printed some details in this to help users track down which test broke. It'd be helpful if you printed the selector and the root's component name